### PR TITLE
add scePsmDrmGetRifKey and psm license struct

### PR DESCRIPTION
**Pull request by EliCrystal2001**
Maybe should goto a separate psmdrm.h, not sure.

### CODE REVIEW

**Comment by Princess-of-Sleeping** [comment]
sha256digest is maybe RSA sign?

**Comment by EliCrystal2001** [comment]
i dont know, i got this struct from NoPsmDrm.

**Comment by d3m3vilurr** [comment]
@frangarcj can you allow to change the license of the nopsmdrm's structure?

**Comment by KuromeSan** [comment]
this struct is wrong, key is only 0x10 bytes, see: https://github.com/KuromeSan/psse-decrypt/blob/master/decrypt.py#L28
it should be like unk[0x20] key[0x10] signature[0x1C0]

also PSM usually uses MD5 not SHA256, so Princess of Sleeping is probably correct about that being wrong.

**Comment by frangarcj** [comment]
> @frangarcj can you allow to change the license of the nopsmdrm's structure?

No problem

File: include/psp2/npdrm.h [hunk 1]
```diff
@@ -37,6 +37,24 @@ typedef struct SceNpDrmLicense { // size is 0x200
...
+typedef struct ScePsmDrmLicense {
+  char magic[0x8];             
+  uint32_t unk1;               
+  uint32_t unk2;               
+  uint64_t aid;                
+  uint32_t unk3;               
+  uint32_t unk4;               
+  uint64_t start_time;         
+  uint64_t expiration_time;    
+  uint8_t act_sha256digest[0x20];
```
[the remark this answers is not in the document]
Comment (KuromeSan) [reply include/psp2/npdrm.h:49]: offical act sha256sum: 76b2cb81a31f07d18afc6ae903742e717510eb642d9fd4e744ce69012113b951 
offical rif 0x30 : E5E2511CF85B3264DB8B1FE30E8075657524BFF26E3F0452C60973566B613AF0

doenst seem like a sha256sum, possibly hmac?

**Comment by SonicMastr** [comment]
> Maybe should goto a separate psmdrm.h, not sure.

I actually think that due to the fact that it's part of a separate ScePsmDrm library, it probably should. I guess that's really up to @d3m3vilurr though

**Comment by EliCrystal2001** [comment]
> > Maybe should goto a separate psmdrm.h, not sure.
> 
> I actually think that due to the fact that it's part of a separate ScePsmDrm library, it probably should. I guess that's really up to @d3m3vilurr though

however its still in NpDrm.skprx.

**Comment by SonicMastr** [comment]
Yes. But having the same header for multiple libraries generally isn't good practice. A module can either be a single library or a collection of libraries and just having them separated would make it more intuitive to use. Not all headers in VitaSDK are named specifically after the module but as the sub-libraries within them to make it easier to differentiate the underlying libraries and functions. It's more of a clarity thing.

File: include/psp2/npdrm.h [hunk 2]
```diff
@@ -72,6 +90,19 @@ int _sceNpDrmGetFixedRifName(char *rif_name, uint64_t aid);
...
+/**
+ * Get PSM rif key
+ *
+ * @param[in] license_buf - RIF buffer (1024 bytes)
+ *
+ * @param[out] keydata - Decrypted key data
+ *
+ * @param[in] flags - Unknown
+ * 
+ * @return >0 on success, <0 on error
```
Comment (d3m3vilurr) [inline include/psp2/npdrm.h:102]: is it right? most time 0 is normal.

Comment (EliCrystal2001) [reply include/psp2/npdrm.h]: your right it is =0.

**Pull request merged**